pkg/endpoint/providers/kube/client.go

```diff
@@ -108,36 +108,34 @@ func (c Client) ListEndpointsForService(svc service.MeshService) []endpoint.Endp
 func (c Client) GetServicesForServiceAccount(svcAccount service.K8sServiceAccount) ([]service.MeshService, error) {
 	log.Info().Msgf("[%s] Getting Services for service account %s on Kubernetes", c.providerIdent, svcAccount)
 	services := mapset.NewSet()
-	podsInterface := c.caches.Pods.List()
 
-	for _, pods := range podsInterface {
-		kubernetesPods := pods.(*corev1.Pod)
-		if kubernetesPods == nil || !c.kubeController.IsMonitoredNamespace(kubernetesPods.Namespace) {
-			// Doesn't belong to namespaces we are observing
+	for _, podInterface := range c.caches.Pods.List() {
+		pod := podInterface.(*corev1.Pod)
+		if pod == nil {
 			continue
 		}
```

Comment:
Don't think a pod can be nil.

Reply:
It could be if `podInterface` is not of type `*corev1.Pod`

Reply:
But that shouldn't happen given the informers are subscribing to pod events using `informerFactory.Core().V1().Pods().Informer()`